internal/helm/registry/auth.go

```diff
@@ -30,6 +47,24 @@ func LoginOptionFromSecret(registryURL string, secret corev1.Secret) (registry.L
 		if err != nil {
 			return nil, fmt.Errorf("unable to get authentication data from Secret '%s': %w", secret.Name, err)
 		}
+
+		// Make sure that the obtained auth config is for the requested host.
+		// When the docker config does not contain the credentials for a host,
+		// the credential store returns an empty auth config.
+		// Refer: https://github.com/docker/cli/blob/v20.10.16/cli/config/credentials/file_store.go#L44
+		serverAddress := authConfig.ServerAddress
+		// If the server address contains scheme, parse and use it to compare
+		// with the registry URL host.
+		if strings.Contains(serverAddress, "//") {
+			serverURL, err := url.Parse(serverAddress)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse server URL '%s' in the docker-registry Secret '%s'", serverAddress, secret.Name)
+			}
+			serverAddress = serverURL.Host
+		}
```

Comment:
I think you should just re-use the func from [here](https://github.com/docker/cli/blob/84b86e23b7f520b0bb956df88331d6c8d472ce0e/cli/config/credentials/file_store.go#L70).

Reply:
Added. Much better now.